src/uct/ib/base/ib_md.c

```diff
@@ -406,8 +421,9 @@ uct_ib_md_handle_mr_list_multithreaded(uct_ib_md_t *md, void *address,
             }
 
             CPU_ZERO(&thread_set);
-            CPU_SET(cpu_id++, &thread_set);
-            pthread_attr_setaffinity_np(&attr, sizeof(cpu_set_t), &thread_set);
+            CPU_SET(cpu_id, &thread_set);
+            cpu_id++;
+            pthread_attr_setaffinity_np(&attr, sizeof(ucs_sys_cpuset_t), &thread_set);
```

Comment:
do we also need to port pthread_get/set_affinity_np?

Reply:
get/set_affinity_np seems to be compatible between FreeBSD and Linux.  I can add a wrapper but lets push this changeset first.

Reply:
ok, if it works on FreeBSD it's not a problem

Reply:
Note the move of cpu_id++ out of CPU_SET macro.  The macro is not referential-transparent in FreeBSD.